src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java

```diff
@@ -51,7 +51,7 @@
 
   public ConnectionPluginManager(Log logger) {
     if (logger == null) {
-      throw new NullArgumentException("logger");
+      throw new RuntimeException(NullArgumentException.getMessage("logger"));
```

Comment:
Please do not throw RuntimeException, it is a very generic exception. 
NullArgumentException extends `IllegalArgumentException`, so it would make a lot more sense to throw IllegalArgumentException instead of RuntimeException.

Reply:
Done.